testsuite/tests/multicluster/conftest.py

```diff
@@ -0,0 +1,184 @@
+"""Conftest for Multicluster tests"""
+
+import time
+from importlib import resources
+from typing import TypeVar
+
+import pytest
+from openshift_client import selector, OpenShiftPythonException
+
+from testsuite.backend.httpbin import Httpbin
+from testsuite.certificates import Certificate
+from testsuite.gateway import Exposer, Gateway, CustomReference, Hostname
+from testsuite.gateway.gateway_api.gateway import KuadrantGateway
+from testsuite.gateway.gateway_api.hostname import DNSPolicyExposer
+from testsuite.gateway.gateway_api.route import HTTPRoute
+from testsuite.openshift.client import OpenShiftClient
+from testsuite.policy import Policy
+from testsuite.policy.dns_policy import DNSPolicy
+from testsuite.policy.tls_policy import TLSPolicy
+
+
+AnyPolicy = TypeVar("AnyPolicy", bound=Policy)
+
+
+def generate_policies(clusters: list[OpenShiftClient], policy: AnyPolicy) -> dict[OpenShiftClient, AnyPolicy]:
+    """Copy policies for each cluster"""
+    return {cluster: policy.__class__(policy.as_dict(), context=cluster.context) for cluster in clusters}
+
+
+@pytest.fixture(scope="module")
+def cluster_issuer(testconfig, hub_openshift, skip_or_fail):
+    """Reference to cluster Let's Encrypt certificate issuer"""
+    testconfig.validators.validate(only="letsencrypt")
+    name = testconfig["letsencrypt"]["issuer"]["name"]
+    kind = testconfig["letsencrypt"]["issuer"]["kind"]
+    try:
+        selector(f"{kind}/{name}", static_context=hub_openshift.context).object()
+    except OpenShiftPythonException as exc:
+        skip_or_fail(f"{kind}/{name} is not present on the cluster: {exc}")
+    return CustomReference(
+        group="cert-manager.io",
+        kind=kind,
+        name=name,
+    )
+
+
+@pytest.fixture(scope="session")
+def openshifts(testconfig, hub_openshift, skip_or_fail) -> list[OpenShiftClient]:
+    """Returns list of all OpenShifts on which to run Multicluster"""
+    additional_clusters = testconfig["control_plane"]["additional_clusters"]
+    if len(additional_clusters) == 0:
+        skip_or_fail("Only one cluster was provided for multi-cluster tests")
+    return [
+        hub_openshift,
+        *(cluster.change_project(testconfig["service_protection"]["project"]) for cluster in additional_clusters),
```

Comment:
I know it's done on multiple places but why do you change project to service_protection.project instead of using the one passed in cluster definition? (what if in a weird scenario one wants to use different namespaces on different clusters?)

Reply:
Cluster project is no longer used (because you need at least 3 namespaces not one) and using different namespaces for each cluster is not supported for same reason

Reply:
if so I'd remove it from settings.local.yaml etc. to avoid confusion (of course in another PR :) ) just came around it and found it a bit weird

Reply:
Done